pkgs/development/interpreters/python/mk-python-derivation.nix

```
@@ -163,15 +157,42 @@ let
   # Python packages built through cross-compilation are always for the host platform.
   disallowedReferences = lib.optionals (python.stdenv.hostPlatform != python.stdenv.buildPlatform) [ python.pythonForBuild ];
 
+  passthru = passthru // {
+    tests.installation = stdenv.mkDerivation {
+      name = "testing-${self.name}";
+      inherit (self) LANG strictDeps src;
+      preUnpack = self.preUnpack or "";
+      postUnpack = self.postUnpack or "";
+      sourceRoot = self.sourceRoot or "";
+
+      installCheckPhase = attrs.checkPhase or ":";
+
+      nativeBuildInputs = [
+        self
+      ] ++ lib.optionals (format == "setuptools") [
+        # Longer-term we should get rid of this and require
+        # users of this function to set the `installCheckPhase` or
+        # pass in a hook that sets it.
+        #setuptoolsCheckHook
+      ] ++ checkInputs ++ propagatedBuildInputs;
```

it might be time to just drop this altogether, already commented it out:
```suggestion
      nativeBuildInputs = [
        self
      ] ++ checkInputs ++ propagatedBuildInputs;
```